### PR TITLE
Hcap 973 capture emails

### PR DESCRIPTION
**Pull request by dbayly-freshworks**
Added a new table, waitlist to capture emails. 
Added new route to add emails to the waitlist.

### CODE REVIEW

File: server/services/participants.js [hunk 1]
```diff
@@ -166,6 +166,19 @@ const withdrawParticipant = async (participantInfo) => {
   return updateParticipant(participant);
 };
 
+const addParticipantToWaitlist = async (email) => {
```
Comment (fwpushan) [inline server/services/participants.js:169]: For now it ok, but this should a new service `waitlistParticipant`.

**Review by fwpushan — DISMISSED** [review]
LGTM, only one suggestion but not major one.

File: client/src/components/participant-form/EmailSubmissionForm.js [hunk 2]
```diff
@@ -51,11 +52,39 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
+const addEmailToWhitelist = async (email, openToast) => {
```
Comment (vyasworks) [inline client/src/components/participant-form/EmailSubmissionForm.js:55]: waitlist typo?

Comment (dbayly-freshworks) [reply client/src/components/participant-form/EmailSubmissionForm.js:55]: Good catch. I will fix this!

**Review by vyasworks — APPROVED** [review]

File: server/db/schema.js [hunk 3]
```diff
@@ -15,6 +15,7 @@ const collections = {
   POST_SECONDARY_INSTITUTIONS: 'post_secondary_institutions',
   COHORTS: 'cohorts',
   COHORT_PARTICIPANTS: 'cohort_participants',
+  WAITLIST: 'waitlist',
```
Comment (arranfw) [inline server/db/schema.js:18]: Suggestion to make this more specific `participantWaitlist` or `PEOIWaitlist`?

File: client/src/components/participant-form/EmailSubmissionForm.js [hunk 4]
```diff
@@ -51,11 +52,39 @@ const useStyles = makeStyles((theme) => ({
...
+    body: JSON.stringify({ email }),
+    method: 'POST',
+  });
+  if (resp.ok) {
+    openToast({
+      status: ToastStatus.Success,
+      message:
+        'Your email has been added to the list. You will be notified when submissions are open.',
+    });
+  } else if (resp.status === 400) {
```
Comment (arranfw) [inline client/src/components/participant-form/EmailSubmissionForm.js:67]: 409 is probably better here

File: client/src/components/participant-form/EmailSubmissionForm.js [hunk 5]
```diff
@@ -51,11 +52,39 @@ const useStyles = makeStyles((theme) => ({
...
+
 export const EmailSubmissionForm = () => {
   const classes = useStyles();
-
+  const { openToast } = useToast();
   const handleSubmit = async (values, { setSubmitting }) => {
-    await new Promise((resolve) => setTimeout(resolve, 500));
+    setSubmitting(true);
+    try {
+      await addEmailToWhitelist(values.email, openToast);
```
Comment (arranfw) [inline client/src/components/participant-form/EmailSubmissionForm.js:81]: 👍

File: client/src/components/participant-form/EmailSubmissionForm.js (hunk 4 above)
Comment (arranfw) [reply client/src/components/participant-form/EmailSubmissionForm.js:67]: Should throw an error in the else for non 200/400 errors so the catch later on is triggered?

File: server/routes/participant.js [hunk 6]
```diff
@@ -111,6 +113,15 @@ participantsRouter.get(
   })
 );
 
+participantsRouter.post(
+  '/waitlist',
+  asyncMiddleware(async (req, res) => {
+    await validate(WaitlistEmailSchema, req.body);
+    return (await addParticipantToWaitlist(req.body.email))
```
Comment (arranfw) [inline server/routes/participant.js:120]: Not a huge fan of this ternary, can we expand it?

**Review by arranfw — APPROVED** [review]

**Pull request merged**